src/array.jl

```diff
@@ -298,6 +298,20 @@ end
         convert{T, N}(::Type{$A{T, N}}, A::$A{T, N}) = A
         convert{T}(::Type{$A{T}}, A::$A{T}) = A
         convert(::Type{$A}, A::$A) = A
+
+
+        ## Other methods
+        function vcat(A::$A...)
+            newlevels, ordered = mergelevels(map(levels, A)...)
+
+            refs = [begin
+                r = a.refs
+                r[r .> 0] = indexin(index(a.pool), newlevels)[r[r .> 0]]
```

Comment:
I don't find multiline comprehensions very easy to read. Also, `r .> 0` is going to allocate a lot: better unroll this operation. So you should allocate `refs` beforehand, and then loop over it.